### PR TITLE
Replace ec2-user with gcp-user (demisto-sdk)

### PR DESCRIPTION
**Pull request by samuelFain**
## Related Issues
relates: [CIAC-8955](https://jira-dc.paloaltonetworks.com/browse/CIAC-8955)

## Description
This PR introduces the following changes:
- Update file paths in demisto-sdk which used `ec2-user` to `gcp-user`.
- Grant correct permissions to commit and push changes in the mock files repo (more information in Jira ticket).

### CODE REVIEW

File: demisto_sdk/commands/test_content/mock_server.py [hunk 1]
```diff
@@ -251,6 +251,9 @@ def push_mock_files(self):
                 f"Failed pushing mock files with error: {exc}", real_time=True
             )
         finally:
+            self.logging_module.debug(
+                "Successfully pushed new/updated mock files to mock git repo.", real_time=True
```
Comment (JasBeilin) [inline demisto_sdk/commands/test_content/mock_server.py:255]: Do we actually know it was successful if it is in the finally section?

**Review by JasBeilin — COMMENTED** [review]
Looks good, just one comment.

File: demisto_sdk/commands/test_content/mock_server.py (hunk 1 above)
Comment (samuelFain) [reply demisto_sdk/commands/test_content/mock_server.py:255]: No, we don't. While there are logs for when an Exception is caught, we can't reliably confirm the success of the push, which might be misleading. Removing this log.

**Review by JasBeilin — APPROVED** [review]
Nice!

**Pull request merged**